### PR TITLE
IDEX-3811: Document workspace REST API

### PR DESCRIPTION
**Pull request by evoevodin**
1. Improve `WorkspaceManager`(docs + new methods structure + tests)
2. Document workspace REST API with swagger annotations

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ок

File: platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java [hunk 1]
```diff
@@ -102,36 +104,195 @@ public void setHooks(WorkspaceHooks hooks) {
...
+     * Gets workspace by name and owner.
+     *
+     * <p>Returned instance always permanent(non-temporary), contains websocket channels
+     * and with either {@link WorkspaceStatus#STOPPED} status or status defined by its runtime(if exists).
+     *
+     * @param name
+     *         the name of the workspace
+     * @param owner
+     *         the owner of the workspace
+     * @return the
```
Comment (akorneta) [inline platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java]: ?

File: platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java [hunk 2]
```diff
@@ -102,36 +104,195 @@ public void setHooks(WorkspaceHooks hooks) {
...
+     * @return the list of workspaces or empty list if user doesn't own any workspace
+     * @throws BadRequestException
+     *         when {@code owner} is null
+     * @throws ServerException
+     *         when any server error occurs while getting workspaces with {@link WorkspaceDao#getByOwner(String)}
+     */
+    public List<UsersWorkspaceImpl> getWorkspaces(String owner) throws ServerException, BadRequestException {
+        requiredNotNull(owner, "Required non-null workspace owner");
+        final Map<String, RuntimeWorkspaceImpl> runtimeWorkspaces = new HashMap<>();
+        for (RuntimeWorkspaceImpl runtimeWorkspace : workspaceRegistry.getByOwner(owner)) {
```
Comment (akorneta) [inline platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java]: mb you should create variable for workspaceRegistry.getByOwner(owner) ?

**Comment by garagatyi** [comment]
ok

**Comment by akorneta** [comment]
ok for me

**Pull request merged**